frontend/public/components/graphs/use-prometheus-poll.ts

```diff
@@ -55,7 +55,7 @@ export const usePrometheusPoll = ({
     return () => {
       clearInterval(interval.current);
     };
-  }, [basePath, defaultQueryName, namespace, numSamples, query, timeout, timeSpan, setData]);
```

Comment:
This looks good as the [React docs omit the state update function from the list of deps](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies), but I agree this is weird behavior from the linter. The `setData` function _technically_ will never change, but I'm guessing whatever shallow comparison it's doing is returning `false`.

Reply:
This is happening because we use `useSafetyFirst` instead of `React.useState`. In other words, if it was `const [data, setData] = React.useState([])`, eslint would not require `setData` to be in the list of dependencies.

Reply:
I'd like to phase out `useSafetyFirst`. If there's an in-flight XHR when the component is unmounted, it would be better to cancel the XHR. @TheRealJon found that it's possible with `fetch` using a `signal` parameter.

The problem with `useSafetyFirst` is that it could potentially mask a memory leak by suppressing the warning from React. If we forget to remove an event listener or cancel a `setInterval`, we wouldn't know because we wouldn't see the `setState` on unmounted component warning.

Reply:
Nice catch though @kyoto! Didn't think of that!